From 9f0a47a74cfc5b5dfbb5b4994c32f6cbedf0f01a Mon Sep 17 00:00:00 2001
From: Zybo PCam port <noreply@local>
Date: Thu, 30 Jul 2026 14:25:00 +0900
Subject: [PATCH] media: xilinx: csi2rxss: set media entity function

Linux 6.12 validates the function of every entity registered in a media
device.  The Xilinx CSI-2 RX subsystem subdevice did not assign one, which
made the composite Xilinx video driver fail while it registered the graph.

Classify CSI-2 RX as the standard V4L2 video-interface bridge.

---
 drivers/media/platform/xilinx/xilinx-csi2rxss.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/xilinx/xilinx-csi2rxss.c b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
index 000000000000..111111111111 100644
--- a/drivers/media/platform/xilinx/xilinx-csi2rxss.c
+++ b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
@@ -1057,6 +1057,7 @@ static int xcsi2rxss_probe(struct platform_device *pdev)
 	subdev->dev = dev;
 	strscpy(subdev->name, dev_name(dev), sizeof(subdev->name));
 	subdev->flags |= V4L2_SUBDEV_FL_HAS_EVENTS | V4L2_SUBDEV_FL_HAS_DEVNODE;
+	subdev->entity.function = MEDIA_ENT_F_VID_IF_BRIDGE;
 	subdev->entity.ops = &xcsi2rxss_media_ops;
 	v4l2_set_subdevdata(subdev, xcsi2rxss);
 
-- 
2.43.0
